2025-05-14 02:42:14,863: Exception on /dashboard [GET]
Traceback (most recent call last):
  File "/home/Tayo9M/venv/lib/python3.9/site-packages/flask/app.py", line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/Tayo9M/venv/lib/python3.9/site-packages/flask/app.py", line 1518, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/Tayo9M/venv/lib/python3.9/site-packages/flask/app.py", line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/Tayo9M/venv/lib/python3.9/site-packages/flask/app.py", line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/home/Tayo9M/venv/lib/python3.9/site-packages/flask_login/utils.py", line 272, in decorated_view
    return func(*args, **kwargs)
  File "/home/Tayo9M/sistem-pencatatan-penginapan-kos/./routes.py", line 351, in dashboard
    return render_template(
  File "/home/Tayo9M/venv/lib/python3.9/site-packages/flask/templating.py", line 147, in render_template
    return _render(
  File "/home/Tayo9M/venv/lib/python3.9/site-packages/flask/templating.py", line 128, in _render
    rv = template.render(context)
  File "/home/Tayo9M/venv/lib/python3.9/site-packages/jinja2/environment.py", line 1295, in render
    self.environment.handle_exception()
  File "/home/Tayo9M/venv/lib/python3.9/site-packages/jinja2/environment.py", line 942, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "/home/Tayo9M/sistem-pencatatan-penginapan-kos/templates/dashboard.html", line 1, in top-level template code
    {% extends "layout.html" %}
  File "/home/Tayo9M/sistem-pencatatan-penginapan-kos/templates/layout.html", line 147, in top-level template code
    {% block content %}{% endblock %}
  File "/home/Tayo9M/sistem-pencatatan-penginapan-kos/templates/dashboard.html", line 64, in block 'content'
    <small class="text-muted d-block mt-2">Tingkat Hunian: {{ occupancy_rate|round(1) }}%</small>
  File "/home/Tayo9M/venv/lib/python3.9/site-packages/jinja2/filters.py", line 1182, in do_round
    return round(value, precision)
TypeError: type Undefined doesn't define __round__ method